src/test/java/uk/gov/hmcts/reform/sendletter/controllers/UpdateIsFailedTest.java

```diff
@@ -0,0 +1,59 @@
+package uk.gov.hmcts.reform.sendletter.controllers;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+import uk.gov.hmcts.reform.sendletter.services.AuthChecker;
+import uk.gov.hmcts.reform.sendletter.services.LetterService;
+
+import java.util.UUID;
+
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest
+public class UpdateIsFailedTest {
+    @Autowired private MockMvc mockMvc;
+
+    @MockBean private LetterService letterService; //NOPMD
+    @MockBean private AuthTokenValidator tokenValidator;
+    @MockBean private AuthChecker authChecker;
+
+    @Test
+    public void should_return_204_on_successful_update() throws Exception {
+        given(tokenValidator.getServiceName(anyString())).willReturn("some-service-name");
+
+        mockMvc.perform(
+            put("/letters/" + UUID.randomUUID().toString() + "/is-failed")
+                .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
+                .header("ServiceAuthorization", "auth-header-value")
+                .content("{\"is_failed\": \"true\"}")
+        ).andExpect(status().isNoContent());
+    }
+
+    @Test
+    public void should_validate_service_token() throws Exception {
+
+        final String serviceToken = "my_service_token";
+
+        mockMvc.perform(
+            put("/letters/" + UUID.randomUUID().toString() + "/printed-at")
+                .contentType(MediaType.APPLICATION_JSON_UTF8_VALUE)
+                .header("ServiceAuthorization", serviceToken)
+                .content("{\"is_failed\": \"true\"}")
+        );
+
+        verify(tokenValidator).getServiceName(serviceToken);
+        verify(authChecker).assertCanUpdateLetter(anyString());
```

Comment:
in general, what are we testing here? assertion of whether service is allowed to update, not hitting letter service or what? because just verifying `getServiceName` is called can be moved to previous method too but _given_ directive is kind of already saying that

Reply:
Agree @doncem Actually I have referred to existing update test case and replicated the same here but I will refactor this so that we have `verify(...)` in the same `should_return_204_on_successful_update` method.